src/main/java/org/jenkinsci/test/acceptance/po/JenkinsLogger.java

```diff
@@ -43,6 +43,15 @@ public boolean isEmpty() {
         return getElement(by.css("#main-panel pre")) == null;
     }
 
+    public String getAllMessages() {
+        open();
+        StringBuilder sb = new StringBuilder();
+        for (WebElement e : all(by.css("#main-panel pre"))) {
+            sb.append(e.getText()).append("\n");
+        }
```

Comment:
could possibly be flaky as IIRC this page is updated by ajax?
So you get the elements at time `X` but when you get their text at time `X'` they may be stale.